### PR TITLE
DS17R2

### PR DESCRIPTION
**Pull request by micahyy**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ x] Keymap/layout (addition or update)
- [ x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/.vscode/settings.json
Comment (waffle87) [inline keyboards/.vscode/settings.json:1]: This file and directory should be removed.

File: keyboards/czmao/ds17r2/keyboard.json
Comment (waffle87) [inline keyboards/czmao/ds17r2/keyboard.json:1]: Please run `qmk format-json -i` on this file.

File: keyboards/czmao/ds17r2/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        { "flags": 2, "matrix": [3, 2], "x": 150, "y": 48 },
+        { "flags": 2, "matrix": [3, 1], "x": 75, "y": 48 },
+
+        { "flags": 2, "matrix": [2, 1], "x": 75, "y": 32 }
+        ],
+        "sleep": true
+    },
+
+    "layouts": {
+      "LAYOUT_numberpad_5x4": {
```
Comment (waffle87) [inline keyboards/czmao/ds17r2/keyboard.json:108]: Would the [default 5x4 numpad layout](https://github.com/qmk/qmk_firmware/tree/master/layouts/default/numpad_5x4) work here?

File: keyboards/czmao/ds17r2/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+// enum custom_keycodes {
+//   QWERTY = SAFE_RANGE,
+//   LOWER,
+//   RAISE
+// };
```
Comment (waffle87) [inline keyboards/czmao/ds17r2/keymaps/default/keymap.c:29]:
```suggestion
// Copyright 2024 QMK
// SPDX-License-Identifier: GPL-2.0-or-later
#include QMK_KEYBOARD_H
```

File: keyboards/czmao/ds17r2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    [0] = LAYOUT_numberpad_5x4(
+//1    2        3        4        
+MO(1), KC_PSLS, KC_PAST, KC_PMNS,
+KC_P7, KC_P8,   KC_P9,   
+KC_P4, KC_P5,   KC_P6,   KC_PPLS,
+KC_P1, KC_P2,   KC_P3,   
+KC_P0,          KC_PDOT, KC_PENT
+    ),
+    [1] = LAYOUT_numberpad_5x4(
+MO(1),   KC_NUM,  MO(2),   KC_CALC,
```
Comment (waffle87) [inline keyboards/czmao/ds17r2/keymaps/default/keymap.c:41]:
```suggestion
KC_TRNS,   KC_NUM,  MO(2),   KC_CALC,
```

File: keyboards/czmao/ds17r2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+RGB_HUD, KC_TRNS, RGB_HUI, RGB_VAD,  
+KC_TRNS, RGB_SAD, KC_TRNS, 
+RGB_MOD,          KC_TRNS, KC_TRNS
+    ),
+        [3] = LAYOUT_numberpad_5x4(
+KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAI,
+KC_TRNS, KC_TRNS, KC_TRNS, 
+KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+KC_TRNS, KC_TRNS, KC_TRNS, 
+KC_TRNS,          KC_TRNS, KC_TRNS
```
Comment (waffle87) [inline keyboards/czmao/ds17r2/keymaps/default/keymap.c:59]: This layer is not accessible.
```suggestion
```

File: keyboards/czmao/ds17r2/readme.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+* Keyboard Maintainer: [CZMAO](https://github.com/smollchungus)
+* Hardware Supported: *CZMAO*
+* Hardware Availability: *CZMAO Keyboards*
+
+Make example for this keyboard (after setting up your build environment):
+
+    make czmao/ds17r2:default
+
+
```
Comment (waffle87) [inline keyboards/czmao/ds17r2/readme.md:24]:
```suggestion

* Keyboard Maintainer: [CZMAO](https://github.com/smollchungus)
* Hardware Supported: DS17R2 PCB
* Hardware Availability: [Taobao](https://item.taobao.com/item.htm?ft=t&id=819874589305&sku_properties=5919063:3266779;122216346:21959;122216808:10491261)

Make example for this keyboard (after setting up your build environment):

    make czmao/ds17r2:default

Flashing example for this keyboard:

    make czmao/ds17r2:default:flash
```

File: keyboards/czmao/ds17r2/keyboard.json [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    "bootloader": "stm32duino",
+    "usb": {
+      "vid": "0x4A14",
+      "pid": "0x4A14",
+      "device_version": "1.0.0"
+    },
+    "features": {
+        "bootmagic": true,
+        "command": true,
+        "console": true,
```
Comment (waffle87) [inline keyboards/czmao/ds17r2/keyboard.json:17]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
The only difference between the `ds17r2` and `ds17r3l` appears to be different layouts? It would be preferable to just add an additional layout option to one keyboard, and remove the other keyboard entirely.

File: keyboards/czmao/ds17r2/keyboard.json (hunk 1 above)
Comment (micahyy) [reply keyboards/czmao/ds17r2/keyboard.json:106]: sorry not the same layout

File: keyboards/.vscode/settings.json
Comment (micahyy) [reply keyboards/.vscode/settings.json:1]: done

File: keyboards/czmao/ds17r2/keyboard.json
Comment (micahyy) [reply keyboards/czmao/ds17r2/keyboard.json:1]: done

File: keyboards/.vscode/settings.json
Comment (waffle87) [reply keyboards/.vscode/settings.json:1]: Doesn't appear to be.

Comment (micahyy) [reply keyboards/.vscode/settings.json:1]: I can't find this file in my fork, could you help delete it ?

**Comment by micahyy** [comment]
> The only difference between the `ds17r2` and `ds17r3l` appears to be different layouts? It would be preferable to just add an additional layout option to one keyboard, and remove the other keyboard entirely.
Different RGB_Matrix, same layout

File: keyboards/.vscode/settings.json
Comment (drashna) [inline keyboards/.vscode/settings.json:1]: This file should be removed.

Comment (micahyy) [reply keyboards/.vscode/settings.json:1]: removed

Comment (micahyy) [reply keyboards/.vscode/settings.json:1]: Removed

**Pull request closed without merging**